src/apps/competitions/tasks.py

```diff
@@ -423,65 +406,29 @@ def unpack_competition(competition_dataset_pk):
             # ---------------------------------------------------------------------
             # Phases
 
-            for index, phase_data in enumerate(competition_yaml.get('phases')):
+            for index, phase_data in enumerate(sorted(competition_yaml.get('phases'), key=lambda x: x.get('index') or 0)):
+                # This normalizes indexes to be 0 indexed but respects the ordering of phase indexes from the yaml if present
                 new_phase = {
                     "index": index,
                     "name": phase_data.get('name'),
-                    "description": phase_data.get('description') if 'description' in phase_data else None,
-                    "start": parser.parse(str(phase_data.get('start'))) if 'start' in phase_data else None,
-                    "end": parser.parse(phase_data.get('end')) if 'end' in phase_data else None,
-                    'max_submissions_per_day': phase_data.get(
-                        'max_submissions_per_day') if 'max_submissions_per_day' in phase_data else None,
-                    'max_submissions_per_person': phase_data.get(
-                        'max_submissions') if 'max_submissions' in phase_data else None,
+                    "description": phase_data.get('description'),
+                    "start": _get_datetime(phase_data.get('start')),
+                    "end": _get_datetime(phase_data.get('end')),
+                    'max_submissions_per_day': phase_data.get('max_submissions_per_day'),
+                    'max_submissions_per_person': phase_data.get('max_submissions'),
                 }
-                if 'execution_time_limit_ms' in phase_data:
-                    new_phase['execution_time_limit'] = phase_data.get('execution_time_limit_ms')
-                if 'max_submissions_per_day' in phase_data or 'max_submissions' in phase_data:
+                execution_time_limit = phase_data.get('execution_time_limit_ms')
```

Comment:
Hmm, should we add `_ms` or remove it to make these the same on both ends?

Reply:
I think having `_ms` on the yaml is a good idea so it is explicit to the users what units the time limit is in. making the field the same name on the backend would probably be nice

Reply:
I think we should leave this how it is for now since going through and changing all of that is outside the scope of this PR.

Reply:
Sounds good, I'll add a Trello item for it